simple/msg_data_integrity.c

```diff
@@ -0,0 +1,717 @@
+/*
+ * Copyright (c) 2013-2014 Intel Corporation.  All rights reserved.
+ *
+ * This software is available to you under the BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AWV
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <getopt.h>
+#include <time.h>
+#include <netdb.h>
+#include <unistd.h>
+
+#include <rdma/fabric.h>
+#include <rdma/fi_errno.h>
+#include <rdma/fi_endpoint.h>
+#include <rdma/fi_cm.h>
+
+#include "shared.h"
+
+static struct cs_opts opts;
+static int max_credits = 128;
+static int credits = 128;
+static char test_name[10] = "custom";
+static struct timespec start, end;
+static void *recv_buf;
+static void *send_buf;
+static size_t buffer_size;
+static size_t next_size = 16;
+static size_t transfer_size;
+
+static struct fi_info *hints;
+
+static struct fid_fabric *fab;
+static struct fid_pep *pep;
+static struct fid_domain *dom;
+static struct fid_ep *ep;
+static struct fid_eq *cmeq;
+static struct fid_cq *rcq, *scq;
+static struct fid_mr *recv_mr;
+static struct fid_mr *send_mr;
+static int common_seed = 7;
+static const char msg_char[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+static int msg_char_length = (sizeof(msg_char)/sizeof(*msg_char)) - 1;
+static int allowed_error_count = 0;
+static int detailed_output = 0;
+static int detailed_error = 0;
+static int error_count = 0;
+static int first_error_byte = -1;
+static int print_error = 0;
+
+static void print_byte_array(char *buffer, char *array, int size)
+{
+	int j = 0;
+	printf("%s = ", buffer);
+	for(j=0; j<size; j++) {
+		printf("%c", array[j]);
+        }
```

Comment:
indenting looks inconsistent here... maybe you're mixing tabs and spaces?  The libfabric/fabtests style is essentially the Linux kernel style, which consistently uses hard tabs with an 8 character spacing.

Also, the spacing on the `for()` line doesn't match the rest of the project.